src/Agent.Service/Windows/AgentService.cs

```diff
@@ -151,7 +151,7 @@ private Process CreateAgentListener()
             string exeLocation = Assembly.GetEntryAssembly().Location;
             string agentExeLocation = Path.Combine(Path.GetDirectoryName(exeLocation), "Agent.Listener.exe");
             Process newProcess = new Process();
-            newProcess.StartInfo = new ProcessStartInfo(agentExeLocation, "run");
+            newProcess.StartInfo = new ProcessStartInfo(agentExeLocation, "run --startuptype windowsservice");
```

Comment:
the restart of service host is failing.

---
In reply to: [122223235](https://github.com/Microsoft/vsts-agent/pull/985#discussion_r122223235) [](ancestors = 122223235)